projects/js-packages/components/components/text/types.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import type React from 'react';
+
+import { VARIANTS_MAPPING } from './constants';
+
+export type SpacingValue = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+export type TextProps = {
+	variant?: keyof typeof VARIANTS_MAPPING;
+	/** margin */
+	m?: SpacingValue;
+	/** margin-top */
+	mt?: SpacingValue;
+	/** margin-right */
+	mr?: SpacingValue;
+	/** margin-bottom */
+	mb?: SpacingValue;
+	/** margin-left */
+	ml?: SpacingValue;
+	/** margin left and right */
+	mx?: SpacingValue;
+	/** margin top and bottom */
+	my?: SpacingValue;
+	/** padding */
+	p?: SpacingValue;
+	/** padding-top */
+	pt?: SpacingValue;
+	/** padding-right */
+	pr?: SpacingValue;
+	/** padding-bottom */
+	pb?: SpacingValue;
+	/** padding-left */
+	pl?: SpacingValue;
+	/** padding left and right */
+	px?: SpacingValue;
+	/** padding top and bottom */
+	py?: SpacingValue;
+	/** HTML Class */
+	className?: string;
+	/** The text itself that will be rendered. */
+	children?: React.ReactNode;
```

Comment:
Saw the point here: https://github.com/Automattic/jetpack/pull/24256#issuecomment-1118551018 ... Which case we could have an empty text?

Reply:
Not sure, but we can definitely make it required.

Reply:
Made it required. Thanks

Reply:
> Which case we could have an empty text?

```jsx
function setTitle() {
  const [ title, setTitle ] = useState();
  return (
    <>
      <Text
        onChange={ setTitle }
        value={ title }
      />

      <Title>{ title }</Text>
    </>
  );
} 
```
Is it a valid case?

Reply:
@retrofox Yes, it's. But in this case, the better is to use `useState(' ')`, not? Mainly to avoid "jumps" at the interface.

Reply:
yes, because you have control... what happens when you cannot control the value. Probably you'll have to add extra code to handle it. In the end, an empty string is a string, is it?

Reply:
Yep! An empty string is a `string`, so it works fine. The point is only if pass it as `undefined`, which exactly the case I'm not seeing here.

Reply:
Empty string should work fine with `children` being required.